business/jaeger.go

```diff
@@ -0,0 +1,261 @@
+package business
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/kiali/kiali/models"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/log"
+	"github.com/kiali/kiali/util/httputil"
+)
+
+type JaegerService struct {
+	auth          config.Auth
+	businessLayer *Layer
+}
+
+func (in *JaegerService) makeRequest(endpoint string, body io.Reader) (response []byte, status int, err error) {
+	response = nil
+	status = 0
+	client, err := in.getClient()
+	if err != nil {
+		return
+	}
+	req, err := http.NewRequest(http.MethodGet, endpoint, body)
+	if err != nil {
+		return
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return
+	}
+	defer resp.Body.Close()
+	response, err = ioutil.ReadAll(resp.Body)
+	status = resp.StatusCode
+	return
+}
+
+func (in *JaegerService) getClient() (client http.Client, err error) {
+	timeout := time.Duration(1000 * time.Millisecond)
+	client = http.Client{}
+	transport, err := httputil.AuthTransport(&in.auth, &http.Transport{})
+	if err != nil {
+		return
+	}
+	client = http.Client{Transport: transport, Timeout: timeout}
+	return
+}
+
+func getTags(tagsJson []interface{}) (tags []models.JaegerKeyValue) {
+	for _, t := range tagsJson {
+		tag, _ := t.(map[string]interface{})
+		tags = append(tags, models.JaegerKeyValue{
+			Key:   tag["key"].(string),
+			Type:  tag["type"].(string),
+			Value: tag["value"].(string),
+		})
+	}
+	return
+}
+
+func getTraceFromJson(data map[string]interface{}) (trace models.JaegerTrace) {
```

Comment:
I think it would be more typesafe (and easier to use/read) to unmarshal using jaeger's model instead of several nested maps. Is there a reason to not do it?

E.g. 

```go
import (
    jaeger "github.com/jaegertracing/jaeger/model/json"
)

// ...

	var traces RequestTrace
	if errMarshal := json.Unmarshal(body, &traces); errMarshal != nil {
		log.Errorf("Error unmarshalling Jaeger response: %s", errMarshal)
		return []jaeger.Trace{}, errMarshal
	}
// ...
```

Reply:
clean and less memory but I understand your point that this is more not readable for developers